server/src/main/java/com/battleships/Server.java

```diff
@@ -0,0 +1,93 @@
+package com.battleships;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+class Server {
+    private final ServerSocket serverSocket;
+    private final ConnectedPlayers connectedPlayers;
+    private static final Logger logger = LogManager.getLogger(Server.class.getName());
+
+    Server() throws IOException {
+        logger.info("Próba odpalenia serwera");
+        this.serverSocket = new ServerSocket(50000);
+        logger.info("Serwer odpalony... :)");
+        connectedPlayers = new ConnectedPlayers();
+        tryAcceptPlayersLoop();
+    }
+
+    private void tryAcceptPlayersLoop() {
+        new Thread(() -> {
+            while (true) {
+                try {
+                    addPlayerToTheGame();
+                } catch (IOException e) {
+                    logger.error("Wystąpił problem przy próbie dodania nowego gracza", e.getStackTrace());
+                }
+            }
+        }).start();
+    }
+
+
+    private void addPlayerToTheGame() throws IOException {
+        Player player = acceptPlayer();
+        assignNameToNewUser(player);
+        logger.info(String.format("Nowy gracz połączył się do serwera: %s", player));
+        player.sendMessage(String.format("Dostałeś nick: %s", player));
+        registerPlayer(player);
+        new Thread(() -> handlePlayerMessagesUntilDisconnected(player)).start();
```

Comment:
addPlayerToTheGame name doesn't suggest that it should start new thread

Reply:
Is it fixed?

Reply:
fixed